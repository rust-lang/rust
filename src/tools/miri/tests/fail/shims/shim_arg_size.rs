fn main() {
    extern "C" {
        // Use the wrong type (ie. not `i32`) for the `c` argument.
        fn memchr(s: *const std::ffi::c_void, c: u8, n: usize) -> *mut std::ffi::c_void;
    }

    unsafe {
        memchr(std::ptr::null(), 0, 0); //~ ERROR: Undefined Behavior: scalar size mismatch
    };
}
