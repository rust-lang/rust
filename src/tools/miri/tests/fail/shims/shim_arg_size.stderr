error: Undefined Behavior: scalar size mismatch: expected 4 bytes but got 1 bytes instead
  --> tests/fail/shims/shim_arg_size.rs:LL:CC
   |
LL |         memchr(std::ptr::null(), 0, 0);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ scalar size mismatch: expected 4 bytes but got 1 bytes instead
   |
   = help: this indicates a bug in the program: it performed an invalid operation, and caused Undefined Behavior
   = help: see https://doc.rust-lang.org/nightly/reference/behavior-considered-undefined.html for further information
   = note: BACKTRACE:
   = note: inside `main` at tests/fail/shims/shim_arg_size.rs:LL:CC

note: some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace

error: aborting due to 1 previous error

